lib/no_brainer/criteria/join.rb

```diff
@@ -16,6 +16,7 @@ def _compile_join_ast(value)
         association = model.association_metadata[k.to_sym]
         raise "`#{k}' must be an association on `#{model}'" unless association
         raise "join() does not support through associations" if association.options[:through]
+        raise "join() does not support polymorphic associations" if association.options[:polymorphic]
```

Comment:
@nviennot what do you think about this? When trying to `join` on a `belongs_to :xxx, polymorphic: true` association, Nobrainer raises an error, while it allows calling `join` on a `has_many :yyy, as: :xxx`.